gdb.cc

```diff
@@ -562,7 +562,17 @@ read_waypoint(gt_waypt_classes_e* waypt_class_out)
 
     garmin_fs_t::set_addr(gmsd, fread_cstr());
 
-    FREAD(buf, 5);				/* instruction depended */
+    FREAD(buf, 1);
+    unsigned int duration = (unsigned int) FREAD_i32;
```

Comment:
Although gdb doesn't use it gbfgetuint32 exists, so maybe just
unsigned int duration = gbfgetuint32(fin);